### ISSUE TITLE
[0.4.11, Scala 3] "Unable to extract varargs arguments" for exported vararg functions

### ISSUE THREAD

**Issue by keynmol** [scala-native/scala-native#3231]
Reproduction: (scala cli)

```scala
//> using nativeVersion "0.4.11"
//> using platform "scalaNative"
import scala.scalanative.unsafe.extern

@extern
object extern_functions:
  def test(bla: Int, args: Any*): Unit = extern

object functions:
  export extern_functions.test
```

Error (compilation:

```
Compiling project (Scala 3.2.2, Scala Native)
[error] ./test.scala:10:26
[error] Unable to extract vararg arguments, varargs to extern methods must be passed directly to the applied function
[error]   export extern_functions.test
[error]                          ^
Error compiling project (Scala 3.2.2, Scala Native)
```

**Comment by keynmol** [comment]
@WojciechMazur I think this regressed, or may be never worked with packages/multiple files, not sure which one

**test.scala**:

```scala
//> using scala 3.3.1
//> using nativeVersion 0.4.16
//> using platform scala-native

package hello.bla
package functions

import scala.scalanative.unsafe.extern

@extern
private[bla] object extern_functions:
  def test(bla: Int, args: Any*): Unit = extern

export extern_functions.*
```

**all.scala**

```scala
package hello.bla
package all

export _root_.hello.bla.functions.test
```

Run: `scala-cli compile all.scala test.scala`

```
> scli compile all.scala test.scala
Warning: setting /private/var/folders/jr/4vy422xj6zs44hsn5d1_b4g40000gn/T/tmp.E0x1OeP0zL as the project root directory for this run.
Compiling project (Scala 3.3.1, Scala Native 0.4.16)
[error] ./all.scala:4:34
[error] Unable to extract vararg arguments, varargs to extern methods must be passed directly to the applied function
[error] export _root_.hello.bla.functions.test
[error]                                  ^
Error compiling project (Scala 3.3.1, Scala Native 0.4.16)
Compilation failed
```

**Comment by WojciechMazur** [comment]
@keynmol Thank you for reproducer, I've stumbled on it recently as well, but had not time to fix it yet. Should be fixes in #3665. Next release with a fix should be available soon (this week, or start of the next one)
